pkg/virt-controller/watch/network.go

```diff
@@ -91,19 +93,46 @@ func trimDoneInterfaceRequests(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInst
 	vm.Status.InterfaceRequests = updateIfaceRequests
 }
 
-func handleDynamicInterfaceRequests(vm *v1.VirtualMachine) {
+func handleDynamicInterfaceRequests(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInstance, clusterConfig *virtconfig.ClusterConfig) error {
 	if len(vm.Status.InterfaceRequests) == 0 {
-		return
+		return nil
+	}
+
+	// When a VMI is created, in some cases if the VM has no interfaces, the default interface is auto added to it.
+	// Since the auto added interface is defined only on the VMI spec, explicitly add it to the VM before hot-plugging another interface.
+	vmTemplateCopy, err := addDefaultInterfaceToVm(vm, vmi, clusterConfig)
+	if err != nil {
+		return err
 	}
 
-	vmTemplateCopy := vm.Spec.Template.Spec.DeepCopy()
 	for i := range vm.Status.InterfaceRequests {
-		vmTemplateCopy = controller.ApplyNetworkInterfaceRequestOnVMISpec(
+		vmTemplateCopy = controller.ApplyNetworkInterfaceRequestOnVMTemplateSpec(
 			vmTemplateCopy, &vm.Status.InterfaceRequests[i])
 	}
 	vm.Spec.Template.Spec = *vmTemplateCopy
 
-	return
+	return nil
+}
+
+func addDefaultInterfaceToVm(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInstance, clusterConfig *virtconfig.ClusterConfig) (*v1.VirtualMachineInstanceSpec, error) {
+	vmTemplateCopy := vm.Spec.Template.Spec.DeepCopy()
```

Comment:
Note that a VM object copy already passed down the stack: `sync() -> handleDynamicInterfaceRequests()-> addDefaultInterfaceToVm()`
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L2647C3-L2647C3

I understand the reasoning to avoid mutating functions, but in this particular case it doesnt go well because there's mutating function involved that forces another copy  (`clusterConfig.SetVMISpecDefaultNetworkInterface`).

This is not ablocker we can improve it later and reduce object cloning around that area.